include/mxnet/tensor_blob.h

```diff
@@ -196,6 +196,10 @@ class TBlob {
     CHECK(mshadow::DataType<DType>::kFlag == type_flag_)
       << "TBlob.get_with_shape: data type do not match specified type."
       << "Expected: " << type_flag_ << " v.s. given " << mshadow::DataType<DType>::kFlag;
+    // if scalar, handle it with special care
+    if (ndim() == 0) {
+      return this->get_with_shape<Device, 2, DType>(mshadow::Shape2(1, 1), stream);
```

Comment:
why is this needed? should the `shape_.FlatTo2D()` handle it, given that the next call already passes `dptr_`?

Reply:
Because this line would crash.
```cpp
shape_[shape_.ndim() - 1],
```

Reply:
Is the stride needed?

Reply:
Without providing the stride, it already looks up the size of this dimension as stride:
`static const int  kSubdim = dimension - 1;`